include/mbedtls/config.h

```diff
@@ -238,6 +238,7 @@
 //#define MBEDTLS_BLOWFISH_ALT
 //#define MBEDTLS_CAMELLIA_ALT
 //#define MBEDTLS_DES_ALT
+//#define MBEDTLS_ECDSA_ALT
```

Comment:
You are providing option to replace three functions, therefore I think it would be more appropriate to use a MBEDTLS__FUNCTION_NAME__ALT type macro for each instead of a single MBEDTLS__MODULE_NAME__ALT type flag.

Reply:
I understand your point, and I will work on it, although I don't see any other way to have an alternative implementation for ECDSA, since all other functions are related to ASN1 decoding, and other modules not related specifically to the cryptographic operation

Reply:
I am not sure I understand. I was suggesting to do the same that you already do, but instead of using a single `MBEDTLS_ECDSA_ALT` macro, use three different flags (`MBEDTLS_ECDSA_SIGN_ALT`, `MBEDTLS_ECDSA_VERIFY_ALT`, `MBEDTLS_ECDSA_GENKEY_ALT`) in the same way you already do, just slicing the guarded code segment to 3 disjunct pieces.

The result would be pretty much the same like for the `MBEDTLS_AES_XXX_ALT` macros, and these would go into the same section in `config.h`.

Would this raise any significant issues for the ECDSA alternative implementation?

Reply:
It wouldn't raise significant issues. What I mean is, that as I see it, and alternative implementation, should implement only these three functions( perhaps without the genkey, and this is why I will do as you suggested), but personally I don't like adding to much configurable options, as it increases maintenence

Reply:
I completely agree that it would be more concise and simple to have only one flag, and I see the benefit of this too. My problem is, that it wouldn't be consistent with our current model of alternative implementations (either replace a whole module or just a single function) and I would prefer to have consistency over simplicity here.